CHANGELOG.md

```diff
@@ -1,5 +1,17 @@
 # Change log
 
+## 4.0.0
+
+### Breaking Changes
+
+- Drop support for `createCachedSelector` calls with multiple arguments
+- Drop legacy `FifoCacheObject`, `FlatCacheObject`, `LruCacheObject` exports
+- `createStructuredSelector` typings generic signature changed. Now supports following optional generics: `SelectorObject`, `State`, (`FirstParameter`)
```

Comment:
You would suggest updating this part to give a clear example of how to fix the code, for example:

TypeScript typings of `createStructuredSelector` have been changed. Explicitly setting `State` and `Parameter` types is no longer required (nor possible), please update the typings as follows if necessary:
```typescript
// Before:
const selector = createStructuredCachedSelector<State, string>({
    x: (state, id) => state.a[id]
  })((state, id) => id);
// After:
const selector = createStructuredCachedSelector({
    x: (state: State, id: string) => state.a[id]
  })((state, id) => id);
```